backend/clubs/views.py

```diff
@@ -2206,6 +2215,167 @@ def get_serializer_class(self):
+        else:
+            return Response(
+                {"detail": f"No tickets of type {type} left!"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+    @action(detail=True, methods=["get"])
+    def tickets(self, request, *args, **kwargs):
+        """
+        Get information about tickets for particular event
+        ---
+        requestBody: {}
+        responses:
+            "200":
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                buyers:
+                                    type: array
+                                    items:
+                                        type: object
+                                        properties:
+                                            fullname:
+                                                type: string
+                                            id:
+                                                type: string
+                                            type:
+                                                type: string
+                                totals:
+                                    type: array
+                                    items:
+                                        type: object
+                                        properties:
+                                            type:
+                                                type: string
+                                            count:
+                                                type: integer
+                                available:
+                                    type: array
+                                    items:
+                                        type: object
+                                        properties:
+                                            type:
+                                                type: string
+                                            count:
+                                                type: integer
+        ---
+        """
+        event = self.get_object()
+        tickets = Ticket.objects.filter(event=event).annotate(
+            fullname=Concat("owner__first_name", Value(" "), "owner__last_name")
+        )
+        types = tickets.values_list("type", flat=True).distinct()
+        totals = []
+        available = []
+
+        for type in types:
+            totals.append({"type": type, "count": tickets.filter(type=type).count()})
+            available.append(
+                {
+                    "type": type,
+                    "count": (tickets.filter(type=type, owner__isnull=True).count()),
+                }
+            )
+
+        buyers = tickets.filter(owner__isnull=False).values("id", "fullname", "type")
+
+        return Response({"totals": totals, "available": available, "buyers": buyers})
```

Comment:
Buyers probably shouldn't be public because we don't want everyone to know who owns tickets. I don't think it's a huge deal but would be weird if I could like track you down and know for certain that you're doing to show. Maybe just show the user their own tickets or something? I like `totals` and `available` though.

Reply:
Never mind, my bad I didn't realize this viewset had a permission class — this is perfect!